lib/srv/server/ec2_watcher.go

```diff
@@ -201,27 +201,29 @@ type EC2ClientGetter func(ctx context.Context, region string, opts ...awsconfig.
 
 // MatchersToEC2InstanceFetchers converts a list of AWS EC2 Matchers into a list of AWS EC2 Fetchers.
 func MatchersToEC2InstanceFetchers(ctx context.Context, matchers []types.AWSMatcher, getEC2Client EC2ClientGetter, discoveryConfigName string) ([]Fetcher, error) {
-	return matchersToEC2InstanceFetchers(ctx, matchers, func(ctx context.Context, region string, assumeRole *types.AssumeRole, opts ...awsconfig.OptionsFn) (ec2.DescribeInstancesAPIClient, error) {
-		if assumeRole != nil {
-			opts = append(opts, awsconfig.WithAssumeRole(assumeRole.RoleARN, assumeRole.ExternalID))
+	return matchersToEC2InstanceFetchers(ctx, matchers, func(ctx context.Context, region string, matcher types.AWSMatcher, opts ...awsconfig.OptionsFn) (ec2.DescribeInstancesAPIClient, error) {
+		if matcher.AssumeRole != nil {
+			opts = append(opts, awsconfig.WithAssumeRole(matcher.AssumeRole.RoleARN, matcher.AssumeRole.ExternalID))
```

Comment:
I realize that this might be something that's already happening, but if a slice is expanded when passed in as a variadic parameter, the slice itself is actually passed in as is without copying, so this append might be modifying the additional capacity of `opts` here, which could be a problem.

Reply:
Nice catch, I'll copy its values into a new slice.